README.md

```diff
@@ -1,180 +1,144 @@
 ```
 
 We use the `--rm` option because the extracted maps are written to the host,
 thus the Docker container is no longer needed.
 
-Once the maps are extracted we can mount the maps from the host into a data-only
-container. The resulting container will be used by the `worldserver`.
-
-```sh
-$ docker create -it --name tc-maps -v ~/WoW/ServerData:/opt/trinitycore/maps trinitycore data
-```
+Once the maps are extracted to the volume, we can use it in the worldserver container.
 
-#### Advanced: Extracting maps into a Docker image
+### help
 
-It is possible to extract maps into a Docker image. This is useful if you
-plan on deploying to a remote server. Like before, assume the WoW client
-is located at `~/WoW/WoW3.3.5a`.
+Run the help command!
 
 ```sh
-$ docker run --name map-container -it -v ~/WoW/WoW3.3.5a:/opt/wow-client trinitycore extract-maps
+docker run --rm -it trinitycore help
 ```
 
-We do __not__ use the `--rm` option in this case because the maps are written
-into the Docker container's file system, not the host's. We also do not mount
-`~/WoW/ServerData` to `/opt/trinitycore/maps`. When the extraction is complete,
-the container will be stopped, but still exists. It can be found with
+# Worldserver and Authserver
+
+The worldserver and authserver both depend on a database connection. Docker's `--link` command effectively exposes the ports of a running container to a new container 
+via environment variables. Therefore, the [database container](db/README.md) must be running before attempting to start either the auth or world servers.
 
-```sh
-$ docker ps -a | grep map-container
-```
+It is assumed that the name of the running db container is `tc-dbserver`.
 
-Now the container can be committed into an image called `trinitycore-maps`.
+The worldserver depends on the extracted maps accessible at `/usr/local/trinitycore/data`. In the below examples, this is provided by the volume we have created 
+and populated above: TC-maps.
+ 
+NOTE: in the following commands, the `-i` is very important when running in daemon mode. Without it, the servers will not actually start or will not later be accessible 
+via interactive prompt using `docker exec`.
 
-```sh
-$ docker commit map-container trinitycore-maps
-```
+## Running the worldserver for the first time.
 
-After `map-container` is committed to an image, the container is not needed and
-can be safely deleted. The resulting image can be pushed to a Docker registry,
-and can be used to create data-only containers to use with `worldserver`.
+As the worldserver will ask if we want to create the database the first time (even with the worldserver.conf Updates section enabled), we will need to connect in interactive mode.
 
 ```sh
-$ docker run -it --name tc-maps -v /opt/trinitycore trinitycore-maps data
+docker run --name tc-worldserverinit --rm -it --link tc-dbserver:TCDB -p 8085:8085 -v TC-maps:/usr/local/trinitycore/data -v TC-config:/usr/local/trinitycore/etc trinitycore worldserver
```

Comment:
The other issue I ran into here was that worldserver was looking for the latest maps (https://github.com/TrinityCore/TrinityCore/releases/download/TDB335.20071/TDB_full_world_335.20071_2020_07_15.7z). What's in the image might be a version or two behind. 

@RezoApio did you ever encounter this? I see you have a comment in the Dockerfile about pulling the latest maps. I wonder if there's a cleaner way we can do this, or maybe we need a script? Is there a way we can pin worldserver to look for a specific sql release?

Reply:
@josefalcon 
not seen this but probably because I run it with no update from the worldserver from the sql file I was using.
I can try to make a scriptable solution to pull always the latest SQL.zip.

Reply:
> probably because I run it with no update from the worldserver from the sql file I was using.

What do you mean by this? Is there a way I can run it without needing the absolute latest world sql files?

Reply:
Re your last remark I am not sure, I was expecting the worldserver code to be kind of agnostic to the version of the world sql but maybe some last minute change with DB modification prevent this.

I have added some code that I need to test that should fixit.
I will have some time tonight for it.

Reply:
Looks like the database is specified at compile-time!

https://github.com/TrinityCore/TrinityCore/blob/6e62bf5a1eebb8e8fc9f59eab7ee590025fc3f31/revision_data.h.in.cmake#L12

But it can be extracted, we have the technology:

```sh
$ TC_DB_VERSION=$(strings worldserver | grep TDB_full)
$ echo $TC_DB_VERSION
TDB_full_world_335.20071_2020_07_15.sql
```

Reply:
An alternative would be to run the world server and if it exits with error, grep the error for `TDB_full*.sql`:

```sh
$ TC_DB_VERSION=$(/usr/local/trinitycore/bin/worldserver | grep -o -E 'TDB_[^"]+' | head -n1)
$ echo $TC_DB_VERSION
TDB_full_world_335.20071_2020_07_15.sql
```

Reply:
Currently trying my new code to find the latest world SQL from the release repo